spec/mem/atomic.md

```diff
@@ -0,0 +1,63 @@
+# MiniRust atomic memory
+
+This is a wrapper for a memory. It is used by the machine to distinguish between non-atomic and atomic memory accesses.
+
+```rust
+pub struct AtomicMemory<M: Memory>{
+    memory: M,
+}
+
+/// The different kind of atomicity
+pub enum Atomicity {
+    Atomic,
+    Default,
+
+    /// Not at run-time
+    None,
```

Comment:
I don't think we want both `Default` and `None`. And `Default` is unclear anyway, so why not just use `Atomicity::None` everywhere that you currently use `Default`?

Reply:
Thinking about it `Default` was a bad name. The thing is I already planned ahead and `None` was meant for when there is no active thread, while `Default` were just the default memory accesses executed by a thread.

Reply:
I see you used `None` for initializing the globals. I think it's fine to just make them non-atomic.